### DISCUSSION TITLE
Criticism of OpenCensus Metrics API

### DISCUSSION THREAD

**Discussion by gilgad13** [open-telemetry/opentelemetry-specification#1718] [General]
From first impressions, I found the metrics API expressed in https://opencensus.io/stats/ to be relatively intuitive.  The metrics API expressed so far in the OpenTelemetry specs are quite different.  I see that the OpenTelemetry API is trying to cover both Callback / "queried" and Observation -style metrics, whereas OpenCensus seems to focus entirely on recording and aggregating observations, but even confined to the Observation / "synchronous" side there are a number of differences.

Some ergonomic issues only show up after extensive use, and there some shared contributors to both projects, so I assume the difference is not an accident.  Is there a write up anywhere of critiques, "non-obvious pitfalls" or other problems / nitpicks with the OpenCensus-style synchronous metrics API?

**Comment by reyang** [comment]
Good question @gilgad13!

The root would always go to **the goal** which is documented here https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/README.md#design-goals.

Some of the discussions can be found https://github.com/open-telemetry/opentelemetry-specification/pull/1617#issue-615476535.
